jina/types/arrays/traversable.py

```diff
@@ -95,3 +97,22 @@ def traverse_flat(self, traversal_paths: Iterable[str]) -> 'TraversableSequence'
     @abstractmethod
     def _flatten(sequence):
         ...
+
+    @staticmethod
```

Comment:
can you simply move `get_docs_batch_generator` function into `.batch` and make `.batch` **not** staticmethod?

The reason is that `DocumentArray` to jina is like `ndarray` to numpy, it is our basic data type, and we'd like to have any method that relate to `DocumentArray` or `DocumentArrayMemmap` as its "native" class method. We do not prefer "misc"/"util" module to store orphan methods, it is not part of our design. In Jina, every method should have a clear parent (at least to the best reasonable effort).